### PR TITLE
Provide https example for nginx configuration

### PR DESCRIPTION
**Pull request by Thanathros**
## Changes in this pull request  
This pull request primarily aims at providing a proper https configuration for nginx. This is related to recent changes to the apache configuration, which adapts LetsEncrypt.

## Additional info  
A few comments were added that might refer to external links. These have been added to better reflect certain behavior.
I have been running this (a bit similar) configuration myself.

### CODE REVIEW

**Comment by Thanathros** [comment]
I am not entirely sure (and haven't found reference to) how pull strategy is. Is it squashed on merge ? Merge while pending ? Just give me a small heads up and I shall make sure it is approperiate :)

File: doc/Development_Documentation/23_Installation_and_Upgrade/03_System_Setup_and_Hosting/02_Nginx_Configuration.md [hunk 1]
```diff
@@ -171,6 +176,279 @@ server {
...
+    # A good reference as explaination can be found here:
+    # https://www.digitalocean.com/community/tutorials/understanding-nginx-server-and-location-block-selection-algorithms#matching-location-blocks
+    location ~* /\.well-known/ {
+      try_files $uri /;
+    }
+
+    # Please note that return is cheaper than redirect    
+    # See: https://www.nginx.com/resources/wiki/start/topics/tutorials/config_pitfalls/#taxing-rewrites
+    location / {
+       return 301 https://pimcore.loc$request_uri;
```
Comment (NiklasBr) [inline doc/Development_Documentation/23_Installation_and_Upgrade/03_System_Setup_and_Hosting/02_Nginx_Configuration.md:212]:
```suggestion
       return 301 https://$host$request_uri;
```

Should be possible here https://nginx.org/en/docs/http/ngx_http_core_module.html#var_host

**Comment by Thanathros** [comment]
Oops, small misclick. Give me a moment to properly fix that commit.

**Comment by brusch** [comment]
Thanks a lot!

**Pull request merged**